src/features/class-rest-api.php

```diff
@@ -23,26 +23,125 @@ public function __construct() {}
 	 * Boot the feature.
 	 */
 	public function boot(): void {
-		add_filter( 'rest_post_query', [ $this, 'add_type_param' ], 10, 2 );
+		add_action( 'rest_api_init', [ $this, 'register_endpoints' ] );
 	}
 
 	/**
-	 * Add post_type to rest post query if the type param is set.
+	 * Sets up the endpoint.
 	 *
-	 * @param array<array<int, string>|string> $query_args The existing query args.
-	 * @param WP_REST_Request                  $request The REST request.
-	 * @return array<array<int, string>|string>
+	 * @return void
 	 */
-	// @phpstan-ignore-next-line
-	public function add_type_param( $query_args, $request ): array { // phpcs:ignore Squiz.Commenting.FunctionComment.WrongStyle
-		$type = $request->get_param( 'type' );
-
-		if ( ! empty( $type ) && is_string( $type ) ) {
-			$types                   = explode( ',', $type );
-			$types                   = array_filter( $types, 'post_type_exists' );
-			$query_args['post_type'] = $types;
+	public function register_endpoints(): void {
+		register_rest_route(
+			'wp-curate/v1',
+			'/posts/',
+			[
+				'methods'             => 'GET',
+				'callback'            => [ $this, 'get_posts' ],
+				'permission_callback' => function () {
+					return current_user_can( 'edit_posts' );
```

Comment:
I wonder if this is too restrictive ... what if a user doesn't have the ability to edit the core posts post type but does have the ability to edit a custom post type? I don't think we're exposing any sensitive data here that wouldn't otherwise be available by the main posts endpoint, which is public, so maybe we make this a simple "is user logged in" check instead?